### PR TITLE
tests/server: reduce test time for heartbeat msg cannot be triggered

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Ref #7969

### What is changed and how does it work?

```commit-message
```

### Performance
can reduce test time from 5s+1min(wait for keepalive) to 5s

### Why change?

Previous test want to mock client send region heartbeat to server, and then server will return msg to client.

But Previous test only can be triggered by keepalive https://github.com/tikv/pd/blob/f51f9134558e793e7ea08fc32cef672c7afa37b7/pkg/schedule/hbstream/heartbeat_streams.go#L154
So I changed region id to 0 which will send error msg https://github.com/tikv/pd/blob/5c58dddc66feb343e19d7f792852bb5b6184d9dd/server/grpc_service.go#L1279-L1284
and then client will get msg by https://github.com/tikv/pd/blob/f51f9134558e793e7ea08fc32cef672c7afa37b7/pkg/schedule/hbstream/heartbeat_streams.go#L127

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/server/cluster/cluster_test.go [hunk 1]
```diff
@@ -753,20 +753,19 @@ func TestConcurrentHandleRegion(t *testing.T) {
...
 		re.NoError(err)
-		regionID, err := id.Alloc()
-		re.NoError(err)
 		peer := &metapb.Peer{Id: peerID, StoreId: store.GetId()}
 		regionReq := &pdpb.RegionHeartbeatRequest{
 			Header: testutil.NewRequestHeader(clusterID),
 			Region: &metapb.Region{
-				Id:    regionID,
+				// mock error msg to trigger stream.Recv()
+				Id:    0,
```
Comment (rleungx) [inline tests/server/cluster/cluster_test.go:761]: Will it skip some processes of handle region heartbeat?

Comment (HuSharp) [reply tests/server/cluster/cluster_test.go:761]: yes, but this `regionReq` wants to ensure the client can get the server's response.

- only call `SendScheduleCommand` will send right msg by `oc.hbStreams.SendMsg(region, cmd)`
- `DispatchFromHeartBeat` will not trigger `SendScheduleCommand`

Comment (rleungx) [reply tests/server/cluster/cluster_test.go:761]: I don't remember what is the purpose of this test. Does it aim to test the concurrent problem or race problem?

Comment (HuSharp) [reply tests/server/cluster/cluster_test.go:761]: Yes, Seems like testing(including race) for concurrent running `HandleRegionHeartbeat`

Comment (HuSharp) [reply tests/server/cluster/cluster_test.go:761]: I think core logic for this test is here 
https://github.com/tikv/pd/pull/7998/files#diff-4f23e8da76548a94e6779db8719386a8eabb85ad56deca5e6271a8e162652baaR789-R818

Comment (rleungx) [reply tests/server/cluster/cluster_test.go:761]: If the heartbeat process exits earlier, will it still take effect?

Comment (HuSharp) [reply tests/server/cluster/cluster_test.go:761]: I think this part is just like a `ping` which wants to ensure the connection between the client and server.
So I think the msg of the response is unimportant :)

And I have added some breakpoints to ensure concurrent region heartbeat has been handled.

**Comment by HuSharp** [comment]
@rleungx @lhy1024 friendly ping :)

**Comment by HuSharp** [comment]
@rleungx  @lhy1024  friendly ping :)

**Pull request closed without merging**